pkg/controllers/workgenerator/controller_test.go

```diff
@@ -148,6 +147,150 @@ func TestGetWorkNamePrefixFromSnapshotName(t *testing.T) {
+						"invalid": "{invalid-json}",
+					},
+				},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		"config map with cluster and namespace scoped data": {
+			uConfigMap: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "ConfigMap",
+					"metadata": map[string]interface{}{
+						"name":      "test-config",
+						"namespace": "default",
+					},
+					"data": map[string]interface{}{
+						"resource":  `{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod", "namespace": "default"}}`,
+						"resource2": `{"apiVersion": "v1", "kind": "ClusterRole", "metadata": {"name": "test-role"}}`,
+					},
+				},
+			},
+			want: []fleetv1beta1.Manifest{
+				{RawExtension: runtime.RawExtension{Raw: []byte(`{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod", "namespace": "default"}}`)}},
+				{RawExtension: runtime.RawExtension{Raw: []byte(`{"apiVersion": "v1", "kind": "ClusterRole", "metadata": {"name": "test-role"}}`)}},
+			},
+			wantErr: false,
+		},
+		"config map with cluster scoped and cross namespaced resources data": {
+			uConfigMap: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "ConfigMap",
+					"metadata": map[string]interface{}{
+						"name":      "test-config",
+						"namespace": "default",
+					},
+					"data": map[string]interface{}{
+						"resource":  `{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod", "namespace": "not-default"}}`,
+						"resource2": `{"apiVersion": "v1", "kind": "ClusterRole", "metadata": {"name": "test-role"}}`,
+					},
+				},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		"config map with valid entries in different order": {
+			uConfigMap: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "ConfigMap",
+					"metadata": map[string]interface{}{
+						"name":      "test-config",
+						"namespace": "default",
+					},
+					"data": map[string]interface{}{
+						"resource2": `{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod1", "namespace": "default"}}`,
+						"resource1": `{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod2", "namespace": "default"}}`,
+					},
+				},
+			},
+			want: []fleetv1beta1.Manifest{
+				{RawExtension: runtime.RawExtension{Raw: []byte(`{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod2", "namespace": "default"}}`)}},
+				{RawExtension: runtime.RawExtension{Raw: []byte(`{"apiVersion": "v1", "kind": "Pod", "metadata": {"name": "test-pod1", "namespace": "default"}}`)}},
+			},
+			wantErr: false,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			got, err := extractResFromConfigMap(tt.uConfigMap)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("extractResFromConfigMap() error = %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
```suggestion
				t.Fatalf("extractResFromConfigMap() error = %v, wantErr %v", err, tt.wantErr)
```
can use fatalf instead